staging/src/k8s.io/sample-apiserver/pkg/admission/wardleinitializer/wardleinitializer.go

```diff
@@ -18,26 +18,33 @@ package wardleinitializer
 
 import (
 	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/server"
+	clientset "k8s.io/sample-apiserver/pkg/client/clientset/internalversion"
 	informers "k8s.io/sample-apiserver/pkg/client/informers/internalversion"
 )
 
 type pluginInitializer struct {
-	informers informers.SharedInformerFactory
+	config *server.Config
 }
 
 var _ admission.PluginInitializer = pluginInitializer{}
 
 // New creates an instance of wardle admission plugins initializer.
-func New(informers informers.SharedInformerFactory) (pluginInitializer, error) {
+func New(c *server.Config) (pluginInitializer, error) {
 	return pluginInitializer{
-		informers: informers,
+		config: c,
```

Comment:
can not initialize informers until LoopbackClientConfig initialized in `RecommendedOptions. ApplyTo`, so must pass server.Config to wardleinitializer and initialize informers in  `pluginInitializer.Initialize`

Reply:
The clean way would be to have a `WantsWardleInformerFactory` interface and an initializer for it.

Reply:
Oh, this is the initializer :)

Reply:
This is really tricky. I do not like having the whole config here. But I don't see a good way to cut the cycle here either. The loopback client always hits us in unexpected ways.

Reply:
I donot like the way passing server.Config to the initializer construct function.